### PR TITLE
Prevent training and non positive numbers

### PR DESCRIPTION
**Pull request by fpopa**
Extracted the validation to a function because it can also be used for the untraining logic.

Added two extra tests for the training method, checking for 0 and negative numbers.

### CODE REVIEW

File: apps/api/src/controllers/training.ts [hunk 1]
```diff
@@ -2,20 +2,39 @@ import { UnitTypes } from '@darkthrone/game-data';
 import { PlayerUnits } from '@darkthrone/interfaces';
 import { Request, Response } from 'express';
 import PlayerUnitsModel from '../models/playerUnits';
+import { APIError } from '@darkthrone/client-library';
+
+const training_request_errors = (desiredUnits: PlayerUnits[]): APIError[] => {
```
Comment (MattGibney) [inline apps/api/src/controllers/training.ts:7]: Prefer CamelCase and rename function to `validateTrainingInputs`

**Review by MattGibney — CHANGES_REQUESTED** [review]
Very minor change requested.

Nice work DRYing up the code 👍🏻

**Review by MattGibney — APPROVED** [review]
LGTM
